### ISSUE TITLE
Rails 5.1.0.rc1 - rails-ujs cancel request when returning to edit page

### ISSUE THREAD

**Issue by benoitongit** [rails/rails#28583]
### Steps to reproduce
Brand new project pulled from 5.1.0.rc1, can be find here:
https://github.com/benoitongit/rails-ujs-issue
Just added an edit and show page and home_controller.rb.

show.html.erb
`<h1>Show</h1>`
`<%= link_to 'Edit', edit_path %>`

edit.html.erb
`<%= form_tag '/test', remote: true do %>`
   `<h1>Submit</h1>`
    `<%= submit_tag 'submit' %>`
`<% end %>`

home_controller.rb
`  def update`
    `redirect_to action: 'show'`
  `end`

### Expected behavior
1/ Click on Edit link -> redirected to edit page
2/ Click on Submit form -> redirected to show page
3/ Click on edit link -> redirected to edit page

### Actual behavior
1/ Click on Edit link -> redirected to edit page
2/ Click on Submit form -> redirected to show page
3/ Click on edit link -> request is cancelled, staying on show page :/

If I replace rails-ujs with jquery-rails gem it then works correctly.

### System configuration
**Rails version**: 5.1.0.rc1

**Ruby version**: 2.4.0

**Comment by Fudoshiki** [comment]
I reported about this issue in rails-ujs gem repository
Rails master issue confirmed

**Comment by Fudoshiki** [comment]
I have this issue too

**Comment by DmytroVasin** [comment]
Faced with same issue

I assume that this bug relates to `turbolinks` cache.
[Turbolinks.clearCache() not working as expected?](https://github.com/turbolinks/turbolinks/issues/87) issue.

You can fix that by adding:
```
<meta name="turbolinks-cache-control" content="no-cache">
```

Also please note if you do next steps you also fix this bug:

> RE: Expected behavior
> 
> 1/ Click on Edit link -> redirected to edit page
> 2/ Click on Submit form -> redirected to show page
-> Call: "Turbolinks.clearCache()" in console.
> 3/ Click on edit link -> redirected to edit page

But i can't understand why app works with:
```
//= require jquery
//= require jquery_ujs
//= require turbolinks
```

But does not work with:
```
//= require rails-ujs
//= require turbolinks
```

**Comment by Fudoshiki** [comment]
same

**Comment by DmytroVasin** [comment]
TL, DR:
This issue has been solved by next PR: [28748](https://github.com/rails/rails/pull/28748)

Issue caused by next:

When you use `turbolinks` it rewrites `redirect_to` method. So, when you type `redirect_to` inside your controller it executes [next lines](https://github.com/turbolinks/turbolinks-rails/blob/master/lib/turbolinks/redirection.rb#L30-L31) of code.

Bug happens because `rails-ujs` does not clear executed script and lefts it on the page.

This issue can be closed.

**Comment by maclover7** [comment]
@benoitongit Just to confirm before closing, this issue was related to scripts sent by the server not being executed on the client, which is solved by #28748.

**Comment by benoitongit** [comment]
Sounds good and makes sense, thanks for letting me know.

**Comment by haggen** [comment]
I'm having this issue right now with Rails 5.1.2. Here's my [Gemlock.file](https://gist.github.com/haggen/c6d88c4b65f571c6b37ea346c7dc61e1).

When I submit the form with both `turbolinks` and `rails-ujs` enabled, a request is made but the page doesn't change. If I disable `rails-ujs` things start to work normal again.

**Comment by DmytroVasin** [comment]
@haggen Can you please create `rails new` project that reproduce this bug?

**Comment by haggen** [comment]
@DmytroVasin Just did and it seemed to work. I actually narrowed down the error; with `rails-ujs` enabled turbolinks does not update the page if the form being submitted rendered the same view (a.k.a. the form has validation errors). Now if the submit ended with a `redirect` or if I remove `rails-ujs` everything works correctly. I've tried undoing all my last 10 or so changes, nothing seems take effect. I don't know what else to do. Anyway, sorry for commenting on the wrong issue.
